cmake/CMakeLists.txt

```diff
@@ -1229,6 +1229,15 @@ if (onnxruntime_USE_VITISAI)
   endif()
 endif()
 
+execute_process(COMMAND git log -1 --format=%h
```

Comment:
Some of our build environments do not have git. So this would not work.

Reply:
@snnn It does not matter whether the build envs have git or not. The cmake command won't fail. Instead, it will output to the variable `ORT_GIT_COMMIT` & `ORT_GIT_BRANCH` with an empty string. To make this clearer, I added a comment here and explicitly change the empty string to "Not-Available". An example is show below:

![image](https://user-images.githubusercontent.com/19584326/233908799-cc69aa24-0f64-43d8-b5bf-662b790ae243.png)

Reply:
@snnn I have reverted the change related to git in `cmake/onnxruntime_config.h.in`. For `GetBuildInfoString` in envs that has no git, it will returns the build type and cxx flags as follows. If the env has git, it will contain more information. Is it better?
I have also reverted the change in `tuning_context_impl.h` since the commit id requires git.
![image](https://user-images.githubusercontent.com/19584326/234164907-0a23ecb7-80b1-4ad5-b8a4-727914a2d4e2.png)